### PR TITLE
Expounding on type's table keys

### PR DESCRIPTION
**Pull request by bgribaudo**
Adding more details and examples to `Type.TableKeys`/`Type.ReplaceTableKeys`.

### CODE REVIEW

File: query-languages/m/type-replacetablekeys.md [hunk 1]
```diff
@@ -13,4 +13,66 @@ Type.ReplaceTableKeys(<b>tableType</b> as type, <b>keys</b> as list) as type
 
 ## About
 
-Returns a new table type with all keys replaced by the specified list of keys.
+Returns a new table type with all keys replaced by the specified list of keys. 
+
+Each key is defined using a record in the following form: 
+```powerquery-m
+[
```
Comment (ehrenMSFT) [inline query-languages/m/type-replacetablekeys.md:20]: This key example is kind of non-standard. What does { text } represent, for example? Not saying the way we've done it is ideal, but usually we'd describe the key format, maybe with a bulleted list of fields, and then show how it works via the examples.

File: query-languages/m/type-replacetablekeys.md [hunk 2]
```diff
@@ -13,4 +13,66 @@ Type.ReplaceTableKeys(<b>tableType</b> as type, <b>keys</b> as list) as type
...
+    Columns = { text }, // list of column names defining the key
+    Primary = logical // indicates whether the key is the primary key
+]
+```
+
+The specified list of keys is validated to ensure that no more than one primary key is defined and that all key column names exist on the table type.
+
+## Example 1
+
+Replaces the key information on a table type.
```
Comment (ehrenMSFT) [inline query-languages/m/type-replacetablekeys.md:30]:
```suggestion
Replace the key information on a table type.
```

File: query-languages/m/type-replacetablekeys.md [hunk 3]
```diff
@@ -13,4 +13,66 @@ Type.ReplaceTableKeys(<b>tableType</b> as type, <b>keys</b> as list) as type
...
+{
+    [Columns = {"ID"}, Primary = true],
+    [Columns = {"FirstName", "LastName"}, Primary = false]
+}
+```
+
+
+## Example 2
+
+Clears the key information previously defined on a table type.
```
Comment (ehrenMSFT) [inline query-languages/m/type-replacetablekeys.md:61]:
```suggestion
Clear the key information previously defined on a table type.
```

**Review by ehrenMSFT — CHANGES_REQUESTED** [review]
See my other comments.

**Comment by bgribaudo** [comment]
@ehrenMSFT, thank you for the feedback and for all you do to further Power Query!

Are the revised record field descriptions more along the lines of what you are thinking?

File: query-languages/m/type-replacetablekeys.md [hunk 4]
```diff
@@ -13,4 +13,62 @@ Type.ReplaceTableKeys(<b>tableType</b> as type, <b>keys</b> as list) as type
 
 ## About
 
-Returns a new table type with all keys replaced by the specified list of keys.
+Returns a new table type with all keys replaced by the specified list of keys. 
+
+Each key is defined using a record in the following form: 
+* `Columns`—a list of the column names that define the key
```
Comment (ehrenMSFT) [inline query-languages/m/type-replacetablekeys.md:19]: For lists like this, we usually use colon followed by a space.

```suggestion
* `Columns`: a list of the column names that define the key
```

The same would apply to the other bullets as well.

**Review by ehrenMSFT — CHANGES_REQUESTED** [review]

**Comment by bgribaudo** [comment]
Thanks, @ehrenMSFT. Updated as suggested.

**Review by ehrenMSFT — APPROVED** [review]

**Comment by JasonWHowell** [comment]
Thank you @bgribaudo for the extra examples!

**Pull request merged**